### PR TITLE
refactor: Landing Page builder component background improvements

### PR DESCRIPTION
**Pull request by aresnik11**
### Overview

Removes all of the modes, variants, and props related to dark mode created for the landing page builder components. Instead we are utilizing the `Background` component and the existing `Color Modes` and get all of this for free 🙃 

### PR Checklist

- [x] Related to designs:
- [x] Related to JIRA ticket: [REACH-1328]
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.97-alpha.8ece02.0<br/>@codecademy/gamut-labs@24.1.1-alpha.8ece02.0<br/>@codecademy/styleguide@47.1.1-alpha.8ece02.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6156592111b54a269b0788fa--gamut-preview.netlify.app>https://6156592111b54a269b0788fa--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6156592111b54a269b0788fa>Deploy Logs</a></p>

**Review by christian-dinh — APPROVED** [review]
ya ok this is rad

We're not planning on letting users show dark mode components on light mode pages right? My understanding was that was the original intent here, ie it wasn't so much that `mode="dark"` indicated that the page was in dark mode, just that "oh here's a dark version of this component in case you need some contrast".

If we're abandoning that in favor a global light/dark mode I'm very in favor of doing so, just want to make sure we're aware that that's what we seem to be doing.

**Comment by aresnik11** [comment]
> ya ok this is rad
> 
> We're not planning on letting users show dark mode components on light mode pages right? My understanding was that was the original intent here, ie it wasn't so much that `mode="dark"` indicated that the page was in dark mode, just that "oh here's a dark version of this component in case you need some contrast".
> 
> If we're abandoning that in favor a global light/dark mode I'm very in favor of doing so, just want to make sure we're aware that that's what we seem to be doing.

@christian-dinh i think the idea is still that specific sections can have contrast/be highlighted and not light/dark mode. that is done in static sites where this component is used by throwing a `Background` around it with the color they chose in contentful (see https://github.com/codecademy-engineering/static-sites/pull/951), but since we're using all gamut components we get it all for free with `Background` and don't have to do any extra work to swap colors accordingly

File: packages/gamut-labs/src/LandingPage/Title.tsx [hunk 1]
```diff
@@ -30,12 +23,11 @@ const titleProps = {
...
-  return (
-    <TitleText
-      maxWidth="58rem"
-      {...titleProps[isPageHeading ? 'heading' : 'subheading']}
-      {...rest}
-    />
-  );
-};
+export const Title: React.FC<TitleProps> = ({ isPageHeading, children }) => (
+  <Text
```
Comment (MattSchiller) [inline packages/gamut-labs/src/LandingPage/Title.tsx:27]: Nice cleanup on implicit return 🙇

**Review by MattSchiller — APPROVED** [review]
Looks good. Love all this cleanup. Keeping our code tuned and sharp!

**Review by dreamwasp — APPROVED** [review]
for free!! amazing, love it

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://615b76fb7424f9166b8d138a--gamut-preview.netlify.app>https://615b76fb7424f9166b8d138a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/615b76fb7424f9166b8d138a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.103-alpha.abb95c.0<br/>@codecademy/gamut-labs@24.4.1-alpha.abb95c.0<br/>@codecademy/styleguide@47.4.1-alpha.abb95c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://615b80f21f5f4a1f1b3d4be6--gamut-preview.netlify.app>https://615b80f21f5f4a1f1b3d4be6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/615b80f21f5f4a1f1b3d4be6>Deploy Logs</a></p>

**Pull request merged**